xgboost_ray/util.py

```diff
@@ -145,3 +145,30 @@ def is_ready(self):
                     self._ready_futures.append(obj)
 
         return not bool(self._pending_futures)
+
+
+def get_current_node_resource_key() -> str:
+    """Get the Ray resource key for current node.
+    It can be used for actor placement.
+    If using Ray Client, this will return the resource key for the node that
+    is running the client server.
+    """
+    current_node_id = ray.get_runtime_context().node_id.hex()
+    for node in ray.nodes():
+        if node["NodeID"] == current_node_id:
+            # Found the node.
+            for key in node["Resources"].keys():
+                if key.startswith("node:"):
+                    return key
+    else:
+        raise ValueError("Cannot found the node dictionary for current node.")
+
+
+def force_on_current_node(task_or_actor):
+    """Given a task or actor, place it on the current node.
+
+    If using Ray Client, the current node is the client server node.
+    """
+    node_resource_key = get_current_node_resource_key()
+    options = {"resources": {node_resource_key: 0.01}}
```

Comment:
Does this do a deep update?

Reply:
No this doesn't do a deep update, but I don't think it should. If we want to force on current node then the task or actor should be placed on the current node, ignoring other custom resource requests. I added a comment in the docstring for this.

Reply:
(We don't use the utility to place training actors or alike)